node/src/neighborhood/node_location.rs

```diff
@@ -1,28 +1,24 @@
 // Copyright (c) 2019, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
 
 use ip_country_lib;
-use ip_country_lib::country_finder::{CountryCodeFinder, COUNTRY_CODE_FINDER};
+use ip_country_lib::country_finder::CountryCodeFinder;
 use std::net::IpAddr;
 
-#[allow(dead_code)]
-#[derive(Clone, Debug, Default, Eq)]
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
 pub struct NodeLocation {
     pub country_code: String,
     pub free_world_bit: bool,
 }
 
-impl PartialEq<Self> for NodeLocation {
-    fn eq(&self, other: &Self) -> bool {
-        self.country_code == other.country_code
-    }
-}
-
-pub fn get_node_location(ip_opt: Option<IpAddr>) -> Option<NodeLocation> {
+pub fn get_node_location(
+    ip_opt: Option<IpAddr>,
+    country_code_finder: &CountryCodeFinder,
```

Comment:
I'm not sure where you needed to pull this variable out through the function's interface. As far as I remember you only supply `&COUNTRY_CODE_FINDER` at all those places. It makes a question if it should be inline or not. I'll leave the decision for you as you understand better the code and you might remember also usages (I assume it'd have been in a test) with a reference to a CountryCodeFinder. The problem is I cannot even imagine where you could and why would you get down to constructing a special CountryCodeFinder of somehow different qualities than what the central, official one offers.

Reply:
Thank you, this could go back to use only `COUNTRY_CODE_FINDER`, because we use everywhere only real data